internal/services/springcloud/spring_cloud_api_portal_resource.go

```diff
@@ -185,6 +185,75 @@ func resourceSpringCloudAPIPortalCreateUpdate(d *pluginsdk.ResourceData, meta in
 	return resourceSpringCloudAPIPortalRead(d, meta)
 }
 
+func resourceSpringCloudAPIPortalUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).AppPlatform.APIPortalClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	springId, err := parse.SpringCloudServiceID(d.Get("spring_cloud_service_id").(string))
+	if err != nil {
+		return err
+	}
+	id := parse.NewSpringCloudAPIPortalID(subscriptionId, springId.ResourceGroup, springId.SpringName, d.Get("name").(string))
+
+	existing, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.ApiPortalName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("retreiving %s: %+v", id, err)
+		}
+	}
+	if utils.ResponseWasNotFound(existing.Response) {
+		return fmt.Errorf("retreiving %s: resource was not found", id)
+	}
+
+	if existing.Properties == nil {
+		return fmt.Errorf("retreiving %s: properties are nil", id)
+	}
+	properties := existing.Properties
+
+	if existing.Sku == nil {
+		return fmt.Errorf("retreiving %s: sku is nil", id)
+	}
+	sku := existing.Sku
+
+	if d.HasChange("gateway_ids") {
+		properties.GatewayIds = utils.ExpandStringSlice(d.Get("gateway_ids").(*pluginsdk.Set).List())
+	}
+
+	if d.HasChange("https_only_enabled") {
+		properties.HTTPSOnly = pointer.To(d.Get("https_only_enabled").(bool))
+	}
+
+	if d.HasChange("public_network_access_enabled") {
+		properties.Public = pointer.To(d.Get("public_network_access_enabled").(bool))
+	}
+
+	if d.HasChange("sso") {
+		properties.SsoProperties = expandAPIPortalSsoProperties(d.Get("sso").([]interface{}))
+	}
+
+	if d.HasChange("instance_count") {
+		sku.Capacity = pointer.To(int32(d.Get("instance_count").(int)))
+	}
+
+	apiPortalResource := appplatform.APIPortalResource{
+		Properties: properties,
+		Sku:        sku,
+	}
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.SpringName, id.ApiPortalName, apiPortalResource)
+	if err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
```

Comment:
The error messaged in the create function also need to be updated.
```suggestion
		return fmt.Errorf("updating %s: %+v", id, err)
	}

	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
		return fmt.Errorf("waiting for update of %s: %+v", id, err)
```